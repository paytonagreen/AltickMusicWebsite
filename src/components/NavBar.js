import React, {useEffect } from 'react';
import { Link } from 'react-router-dom';


const NavBar = () => {

  useEffect(() => {
  const navbar = document.querySelector('header');
  let scrolled = false;
    
    window.onscroll = () => {
      if(window.pageYOffset > 100) {
        navbar.classList.add('scrolled');
        if (!scrolled) {
          navbar.style.transform = 'translateY(-70px)';
        }
        setTimeout(() => {
          navbar.style.transform = 'translateY(0px)';
          scrolled = true;
        }, 200)
      } else {
        navbar.classList.remove('scrolled');
        scrolled = false;
      }
    };
  });

  return (
    <div>
      <header>
        <div className="flex-container container">
            <Link className="home-link" to="/"><h3 className="logo"><span className="secondary-text">DL Altick</span> Restoration & Repair</h3></Link>
            <nav>
                {/* <button onClick={() => setHamburgerOpen(!hamburgerOpen)} id="hamburger-button"><div/><div/><div/></button> */}
                  <div className="navbar">
                      <ul>
                          <Link className="restoration-link" to="/restoration"><li>Restoration</li></Link>
                          <Link className="repair-link" to="/repair"><li>Repair</li></Link>
                          <Link className="store-link" to="/store-placeholder"><li>For Sale</li></Link>
                      </ul>
                  </div>
            </nav>
        </div>
      </header>
    </div>
  )
}

export default NavBar;