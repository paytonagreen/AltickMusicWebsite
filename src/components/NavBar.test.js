import React from 'react';
import {shallow} from 'enzyme';
import NavBar from './NavBar.js';


describe('NavBar', () => {
  const navbar = shallow(<NavBar/>)

  it('renders properly', () => {
    expect(navbar).toMatchSnapshot();
  });

  describe('the navbar links', () => {

    it('contain a link to the home page', () => {
      expect(navbar.find('.home-link').props().to).toBe('/')
    });
    it('contain a link to the restoration page', () => {
      expect(navbar.find('.restoration-link').props().to).toBe('/restoration')
    });
    it('contain a link to the repair page', () => {
      expect(navbar.find('.repair-link').props().to).toBe('/repair')
    });
    it('contain a link to the store page', () => {
      expect(navbar.find('.store-link').props().to).toBe('/store-placeholder')
    });
  });
});