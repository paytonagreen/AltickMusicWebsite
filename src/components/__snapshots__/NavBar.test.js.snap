// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders properly 1`] = `
<div>
  <header>
    <div
      className="flex-container container"
    >
      <Link
        className="home-link"
        to="/"
      >
        <h3
          className="logo"
        >
          <span
            className="secondary-text"
          >
            DL Altick
          </span>
           Restoration & Repair
        </h3>
      </Link>
      <nav>
        <div
          className="navbar"
        >
          <ul>
            <Link
              className="restoration-link"
              to="/restoration"
            >
              <li>
                Restoration
              </li>
            </Link>
            <Link
              className="repair-link"
              to="/repair"
            >
              <li>
                Repair
              </li>
            </Link>
            <Link
              className="store-link"
              to="/store-placeholder"
            >
              <li>
                For Sale
              </li>
            </Link>
          </ul>
        </div>
      </nav>
    </div>
  </header>
</div>
`;
